### ISSUE TITLE
Can't insert data when using `source` option for schema field

### ISSUE THREAD

**Issue by hkrutzer** [plausible/ecto_ch#135]
I'm using the `source` field option to map a field to a differently named table column. This works when selecting data, but not when inserting data:

Test:

```elixir
  describe "issue" do
    setup do
      query!("""
      create table if not exists field_source (
        id String,
        field_name String
      ) engine MergeTree
      ORDER BY (id)
      """)

      query!("""
      INSERT INTO field_source VALUES ('insert', 'value');
      """)
      :ok
    end

    defmodule Stuff do
      use Ecto.Schema

      @primary_key false
      schema "field_source" do
        field(:id, Ch, type: "String")
        field(:name, Ch, type: "String", source: :"field_name")
      end
    end

    test "foobar" do
      all(Stuff)
      |> dbg()

      insert_all(Stuff, [
        %{id: "foo", name: "bar"},
        %{id: "foobar", name: "bax"},
      ])
      |> dbg()
    end
  end
```

Output
```
[test/ch/type_test.exs:767: Ch.TypeTest."test issue foobar"/1]
all(Stuff) #=> [
  %Ch.TypeTest.Stuff{
    __meta__: #Ecto.Schema.Metadata<:loaded, "field_source">,
    id: "insert",
    name: "value"
  }
]

  1) test issue foobar (Ch.TypeTest)
     test/ch/type_test.exs:765
     ** (RuntimeError) missing type for field :field_name
     code: insert_all(Stuff, [
     stacktrace:
       (ecto_ch 0.3.0) lib/ecto/adapters/clickhouse/schema.ex:229: anonymous fn/2 in Ecto.Adapters.ClickHouse.Schema.extract_types/2
       (elixir 1.15.7) lib/enum.ex:1693: Enum."-map/2-lists^map/1-1-"/2
       (elixir 1.15.7) lib/enum.ex:1693: Enum."-map/2-lists^map/1-1-"/2
       (ecto_ch 0.3.0) lib/ecto/adapters/clickhouse/schema.ex:44: Ecto.Adapters.ClickHouse.Schema.insert_rows/5
       (ecto_ch 0.3.0) lib/ecto/adapters/clickhouse/schema.ex:33: Ecto.Adapters.ClickHouse.Schema.insert_all/8
       (ecto 3.10.3) lib/ecto/repo/schema.ex:59: Ecto.Repo.Schema.do_insert_all/7
       test/ch/type_test.exs:769: (test)
```

**Comment by ruslandoga** [comment]
👋 @hkrutzer 

The error comes from https://github.com/plausible/ecto_ch/blob/36348a1a838f74efadd8346d0982c39e3079955a/lib/ecto/adapters/clickhouse/schema.ex#L229

Seems like we'd need to check for `__schema__(:field_source, field)` as well.

**Comment by hkrutzer** [comment]
Not sure, I don't know how other adapters do this, but I couldn't easily find it. I guess they don't have to look up the type like ecto_ch has to?

**Comment by ruslandoga** [comment]
Yes, other adapters don't need to do this kind of reflection.

**Comment by ruslandoga** [comment]
https://github.com/plausible/ecto_ch/pull/137 should fix it. It's included in [`v0.3.1`](https://github.com/plausible/ecto_ch/blob/master/CHANGELOG.md#031-2023-11-11) release.

**Comment by hkrutzer** [comment]
It's fixed, thanks very much!
